cmd/xl-storage.go

```diff
@@ -1466,32 +1504,14 @@ func (s *xlStorage) ReadFileStream(ctx context.Context, volume, path string, off
 		return nil, errIsNotRegular
 	}
 
-	if _, err = file.Seek(offset, io.SeekStart); err != nil {
-		return nil, err
-	}
-
-	atomic.AddInt32(&s.activeIOCount, 1)
-	r := struct {
-		io.Reader
-		io.Closer
-	}{Reader: io.LimitReader(file, length), Closer: closeWrapper(func() error {
-		atomic.AddInt32(&s.activeIOCount, -1)
-		return file.Close()
-	})}
-
-	// Add readahead to big reads
-	if length >= readAheadSize {
-		rc, err := readahead.NewReadCloserSize(r, readAheadBuffers, readAheadBufSize)
-		if err != nil {
-			r.Close()
+	if offset > 0 {
```

Comment:
Why is readahead removed from this?

Reply:
Because we can use a common implementation of reading single 2mib buffer for reads.

I found that it's better anyways in the tests.

Reply:
Because we didn't need multiple different implementations to do the same thing @klauspost 
2MiB buffer is sufficient for large objects.